vendor/src/github.com/docker/engine-api/types/container/network_policy_attribute.go

```diff
@@ -0,0 +1,224 @@
+package container
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+const (
+	NetworkFirewallAttrType       = "firewall"
+	NetworkBandwidthAttrType      = "bandwidth"
+	NetworkCOSAttrType            = "cos"
+	NetworkVendorSpecificAttrType = "vendor-specific"
+)
+
+var AllowedPolicyAttributes = map[string]bool{
+	NetworkFirewallAttrType:       true,
+	NetworkBandwidthAttrType:      true,
+	NetworkCOSAttrType:            true,
+	NetworkVendorSpecificAttrType: true,
+}
+
+func InvalidPolicyAttributeStringError(rcvd, exptd string) error {
+	return fmt.Errorf("invalid policy attribute string: %q. Expected: %q", rcvd, exptd)
+}
+
+type firewallRule struct {
+	action   string
+	protocol string
+	port     string
+}
+
+// NetworkFirewallAttr defines a security policy that is collection of
+// actions for traffic matching L3/L4 protocol and L4 port
+type NetworkFirewallAttr struct {
+	rules []firewallRule
```

Comment:
@jainvipin 

looking at the new gist I have a clarification. Is group-id identifying a given set of fw-rules? If so, then may be the groupId info might be redundant here since the policy engine shall be able to push same set of fw-rules for all containers in a group, right? 

Initially I was considering fw-rule as a regular 5-tupple sort of acl-rule that a policy-engine will derive from a logical construct like endpoint-group and push down to the driver but may be you are thinking of group based policy here, in which case putting it alongside the 5-tupple rule(s) seems to violate that indirection. I feel we should be sending either one of these i.e. group-id translated to fw-rules or just group-id but not both to the driver. WDYT?

I think same might hold true for PeerGroupId as well.